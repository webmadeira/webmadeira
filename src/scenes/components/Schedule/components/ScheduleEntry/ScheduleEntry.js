import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { Subtitle2 } from '../../../../../components/typography'
import EntryContent from './components/EntryContent/EntryContent'

const ListItem = styled.li`
  list-style: none;
  display: flex;

  :first-child .entry-info::before {
    top: 3px;
  }

  :last-child .entry-info::before {
    bottom: 42px;
  }
`

const TimeContainer = styled.div`
  padding-right: 45px;
  
  ${({ theme }) => theme.mixins.mediaQuery.phone`
    padding-right: 20px;
  `};  
`

const EntryInfo = styled.div`
  position: relative;
  padding-left: 45px;
  text-align: left;

  ${({ theme }) => theme.mixins.mediaQuery.phone`
    padding-left: 20px;
  `};  

  ::before {
    position: absolute;
    top: 0px;
    bottom: 0;
    left: 0;
    display: block;
    width: 2px;
    content: "";
    background-color: ${props => props.theme.color('neutral', 'bright')};
  }
`

const ScheduleEntry = ({
  entry: {
    title,
    type,
    speakers,
    description,
    length,
    qaLength,
    time,
  },
}) => (
  <ListItem>
    <TimeContainer>
      <Subtitle2>{time}</Subtitle2>
    </TimeContainer>
    <EntryInfo className="entry-info">
      <EntryContent
        title={title}
        type={type}
        speakers={speakers}
        description={description}
        length={length}
        qaLength={qaLength}
      />
    </EntryInfo>
  </ListItem>
)

ScheduleEntry.propTypes = {
  entry: PropTypes.shape({
    title: PropTypes.string.isRequired,
    speakers: PropTypes.arrayOf(PropTypes.string),
    description: PropTypes.string,
    length: PropTypes.number,
    qaLength: PropTypes.number,
  }).isRequired,
}

export default ScheduleEntry
