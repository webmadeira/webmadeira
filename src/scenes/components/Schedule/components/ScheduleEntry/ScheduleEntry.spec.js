import React from 'react'
import { shallow } from 'enzyme'
import ScheduleEntry from './ScheduleEntry'

describe('ScheduleEntry component', () => {
  const entry = {
    title: 'Title',
    speaker: 'Speaker name',
    description: 'Some description',
    length: 20,
    qaLength: 5,
    time: new Date('2018-08-10 15:20:00').toLocaleTimeString('us-US', {
      hour12: false,
      hour: 'numeric',
      minute: 'numeric',
    }),
  }

  it('must render properly', () => {
    const component = shallow(
      <ScheduleEntry
        entry={entry}
      />,
    )

    expect(component).toMatchSnapshot()
  })
})
