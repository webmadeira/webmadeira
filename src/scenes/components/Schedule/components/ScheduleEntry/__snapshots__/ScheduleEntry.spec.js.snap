// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleEntry component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScheduleEntry
    entry={
      Object {
        "description": "Some description",
        "length": 20,
        "qaLength": 5,
        "speaker": "Speaker name",
        "time": "15:20",
        "title": "Title",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <styled.h4>
            15:20
          </styled.h4>
        </styled.div>,
        <styled.div
          className="entry-info"
        >
          <EntryContent
            description="Some description"
            length={20}
            qaLength={5}
            speakers={undefined}
            title="Title"
            type={undefined}
          />
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.h4>
            15:20
          </styled.h4>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "15:20",
          },
          "ref": null,
          "rendered": "15:20",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <EntryContent
            description="Some description"
            length={20}
            qaLength={5}
            speakers={undefined}
            title="Title"
            type={undefined}
          />,
          "className": "entry-info",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Some description",
            "length": 20,
            "qaLength": 5,
            "speakers": undefined,
            "title": "Title",
            "type": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <styled.h4>
              15:20
            </styled.h4>
          </styled.div>,
          <styled.div
            className="entry-info"
          >
            <EntryContent
              description="Some description"
              length={20}
              qaLength={5}
              speakers={undefined}
              title="Title"
              type={undefined}
            />
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.h4>
              15:20
            </styled.h4>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "15:20",
            },
            "ref": null,
            "rendered": "15:20",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <EntryContent
              description="Some description"
              length={20}
              qaLength={5}
              speakers={undefined}
              title="Title"
              type={undefined}
            />,
            "className": "entry-info",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "description": "Some description",
              "length": 20,
              "qaLength": 5,
              "speakers": undefined,
              "title": "Title",
              "type": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
